### PR TITLE
8.1.0-alpha.1

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW

# Summary
- This PR is for a first alpha release of the soe_paragraphs module.

# Review By (Date)
- 6/12/2020, before the tagging party.

# Criticality
- This is an initial release

### CODE REVIEW

File: lib/scss/components/cta-list/_cta-list.scss [hunk 1]
```diff
@@ -0,0 +1,112 @@
+@charset "UTF-8";
+
+.su-cta-list-paragraph{
```
Comment (sherakama) [inline lib/scss/components/cta-list/_cta-list.scss:3]:
```suggestion
.su-cta-list-paragraph {
```

Should have a space between statement and brace.

File: lib/scss/components/cta-list/_cta-list.scss [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+.su-cta-list-paragraph{
+
+    @include modular-spacing('padding-top', 3);
+    @include modular-spacing('padding-left', 2);
+    @include modular-spacing('padding-right', 2);
+
+    border-top: 1px solid $su-color-cool-grey-25;
+
+    @include grid-media-between('xs', 'md'){
```
Comment (sherakama) [inline lib/scss/components/cta-list/_cta-list.scss:11]: `xs` == 0 
So this could be 
```suggestion
    @include grid-media-max('md'){

```

File: lib/scss/components/cta-list/index.scss [hunk 3]
```diff
@@ -0,0 +1,7 @@
+@charset 'UTF-8';
+
+@import
+'decanter/core/src/scss/decanter-no-markup',
```
Comment (sherakama) [inline lib/scss/components/cta-list/index.scss:4]:
```suggestion
```
This is already included in the config.scss file.

File: lib/scss/components/cta-list/index.scss [hunk 4]
```diff
@@ -0,0 +1,7 @@
+@charset 'UTF-8';
+
+@import
+'decanter/core/src/scss/decanter-no-markup',
+'../../config.scss',
+'cta-list.scss';
```
Comment (sherakama) [inline lib/scss/components/cta-list/index.scss:6]:
```suggestion
'../../config',
'cta-list';
```
extensions are not allowed.

File: lib/scss/components/image-cta/_image_cta.scss [hunk 5]
```diff
@@ -0,0 +1,52 @@
+.ptype-image-cta {
```
Comment (sherakama) [inline lib/scss/components/image-cta/_image_cta.scss:1]:
```suggestion
@charset 'UTF-8';

.ptype-image-cta {

```

File: lib/scss/components/image-cta/_image_cta.scss [hunk 6]
```diff
@@ -0,0 +1,52 @@
+.ptype-image-cta {
+    @include grid-media-between('xs', 'sm'){
+        margin-bottom: 0px !important;
```
Comment (sherakama) [inline lib/scss/components/image-cta/_image_cta.scss:3]: Tell me more about this important statement.

**Comment by sherakama** [comment]
I'm pretty sure that code climate only runs on PRs to the base branch so we are not seeing it here but you can see all the issues it found at: https://codeclimate.com/github/SU-SOE/soe_paragraphs/issues

TLDR; 214 items.

File: modules/soe_paragraph_cta_list/config/install/core.entity_form_display.paragraph.cta_list.default.yml [hunk 7]
```diff
@@ -0,0 +1,46 @@
+uuid: 18bddd88-f407-4c08-aad0-3f792823976f
```
Comment (sherakama) [inline modules/soe_paragraph_cta_list/config/install/core.entity_form_display.paragraph.cta_list.default.yml:1]: No UUIDs allowed in the module.

File: modules/soe_paragraph_cta_list/config/install/paragraphs.paragraphs_type.cta_list.yml [hunk 8]
```diff
@@ -0,0 +1,14 @@
+uuid: 720f8414-f928-430d-b01f-5691a02dcf56
+langcode: en
+status: true
+dependencies:
+  content:
+    - 'file:file:c3718e90-6b13-40c4-961a-bb0a28f6ad1c'
+_core:
+  default_config_hash: '-PxPJz2He0UqrQXYVGm9QVGMw9nrviZLOmcSPaE4_Bw'
+id: cta_list
```
Comment (sherakama) [inline modules/soe_paragraph_cta_list/config/install/paragraphs.paragraphs_type.cta_list.yml:9]: Paragraphs should follow the naming convention with prefix `stanford_`
https://docs.google.com/spreadsheets/u/1/d/1ATP3boyFio_s2BYjvzBoOt29p7aQrQV3HB-kZOZwp0g/edit#gid=0

File: composer.json [hunk 9]
```diff
@@ -1,11 +1,14 @@
...
-  "homepage": "https://github.com/SU-SWS/soe_paragraphs",
+  "homepage": "https://github.com/SU-SOE/soe_paragraphs",
   "authors": [],
   "license": "GPL-2.0+",
   "minimum-stability": "dev",
   "prefer-stable": true,
-  "require": {}
+  "require": {
+    "drupal/paragraphs": "~1.8",
+    "su-sws/jumpstart_ui": "dev-8.x-1.x||*"
```
Comment (sherakama) [inline composer.json:12]: Why `||*`?

File: modules/soe_paragraph_cta_list/config/install/core.entity_view_display.paragraph.cta_list.default.yml [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+          variant: default
+    regions:
+      header:
+        - su_cta_list_header
+      deck:
+        - su_cta_list_deck
+      links:
+        - su_cta_list_links
+_core:
+  default_config_hash: PLmkh9jzI6OU7Zi057Qf3WyUwD-vIVaEe0MnzqBfDms
```
Comment (sherakama) [inline modules/soe_paragraph_cta_list/config/install/core.entity_view_display.paragraph.cta_list.default.yml:34]: These statements should be removed from the module.

File: modules/soe_paragraph_cta_list/config/install/field.field.paragraph.cta_list.su_cta_list_deck.yml [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+  config:
+    - field.storage.paragraph.su_cta_list_deck
+    - paragraphs.paragraphs_type.cta_list
+  module:
+    - allowed_formats
+    - text
+third_party_settings:
+  allowed_formats:
+    stanford_html: stanford_html
+    stanford_minimal_html: stanford_minimal_html
```
Comment (sherakama) [inline modules/soe_paragraph_cta_list/config/install/field.field.paragraph.cta_list.su_cta_list_deck.yml:14]: These come from https://github.com/SU-SWS/stanford_text_editor. Be sure to add to `composer.json` and `info.yml` dependencies.

File: modules/soe_paragraph_cta_list/templates/cta-list-paragraph.html.twig [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+ * Available variables:
+ *
+ * {{ header }}
+ * {{ deck }}
+ * {{ links }}
+ *
+ */
+#}
+{%- set attributes = attributes.addClass(['su-cta-list-paragraph', modifier_class]) -%}
+{%- set link_count = links.su_cta_list_links['#items']|length -%}
```
Comment (sherakama) [inline modules/soe_paragraph_cta_list/templates/cta-list-paragraph.html.twig:18]: Patterns are supposed to be generic to any entity type and should not have references to specific fields in them. That requirement is going to make this approach much harder. The `{{ link }}` variable is synonymous with a layout region and could have just about anything in it. A different approach is going to be needed with either a pre-process or field formatted prior to hitting the template.

File: modules/soe_paragraph_cta_list/templates/cta-list-paragraph.html.twig [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+{# We need to figure out how many links go in each column. #}
+{%- if link_count%2 == 0 -%}
+    {%- set col_1_link_count = link_count/2 -%}
+{%- else -%}
+    {%- set col_1_link_count = ((link_count/2)-(1/2))+1 -%}
+{%- endif -%}
+
+<div{{ attributes }}>
+    {{ header }}
+    {{ deck }}
```
Comment (sherakama) [inline modules/soe_paragraph_cta_list/templates/cta-list-paragraph.html.twig:28]: If the regions don't have a wrapper around them could they be consolidated into one?

File: modules/soe_paragraph_image_cta/templates/image-cta-paragraph.html.twig [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+ * {{ image }}
+ * {{ link }}
+ *
+ */
+#}
+
+{%- set attributes = attributes.addClass(['su-image-cta-paragraph', modifier_class]) -%}
+
+<div{{ attributes }}>
+    {# kint() #}
```
Comment (sherakama) [inline modules/soe_paragraph_image_cta/templates/image-cta-paragraph.html.twig:20]: Please remove debugging code prior to review.

File: modules/soe_paragraph_image_cta/templates/image-cta-paragraph.html.twig [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+
+    {{ image }}
+
+    {{ link }}
```
Comment (sherakama) [inline modules/soe_paragraph_image_cta/templates/image-cta-paragraph.html.twig:24]: Can these be consolidated into one region?

File: modules/soe_paragraph_stories/config/install/auto_entitylabel.settings.paragraph.soe_stories.yml [hunk 16]
```diff
@@ -0,0 +1,6 @@
+status: 0
```
Comment (sherakama) [inline modules/soe_paragraph_stories/config/install/auto_entitylabel.settings.paragraph.soe_stories.yml:1]: auto_entitylabel can be removed from this module.

Comment (sherakama) [reply modules/soe_paragraph_stories/config/install/auto_entitylabel.settings.paragraph.soe_stories.yml:1]: I don't think it is needed unless you know something I dont.

File: modules/soe_paragraph_stories/config/install/paragraphs.paragraphs_type.soe_stories.yml [hunk 17]
```diff
@@ -0,0 +1,10 @@
+uuid: 6e81ffa8-9797-4da8-851f-00c6a4827502
+langcode: en
+status: true
+dependencies: {  }
+id: soe_stories
```
Comment (sherakama) [inline modules/soe_paragraph_stories/config/install/paragraphs.paragraphs_type.soe_stories.yml:5]: This should be `stanford_stories` to follow our naming conventions.

File: modules/soe_paragraph_stories/soe_paragraph_stories.info.yml [hunk 18]
```diff
@@ -0,0 +1,12 @@
...
+description: 'A paragraph type to display a person to highlight'
+type: module
+core_version_requirement: ^8.8 || ^9
+package: Stanford
+dependencies:
+  - ds
+  - field
+  - paragraphs
+  - text
+  - soe_paragraphs
```
Comment (sherakama) [inline modules/soe_paragraph_stories/soe_paragraph_stories.info.yml:11]:
```suggestion
  - ds:ds
  - drupal:field
  - paragraphs:paragraphs
  - drupal:text
  - soe_paragraphs:soe_paragraphs
```

In D8/D9 the dependencies are namespace.

**Comment by sherakama** [comment]
@imonroe let me know if you have any questions or comments.  You can also push your code as you go and I can review it asynchronously.

File: modules/soe_paragraph_cta_list/config/install/field.field.paragraph.cta_list.su_cta_list_deck.yml (hunk 11 above)
Comment (imonroe) [reply modules/soe_paragraph_cta_list/config/install/field.field.paragraph.cta_list.su_cta_list_deck.yml]: corrected.

File: modules/soe_paragraph_cta_list/templates/cta-list-paragraph.html.twig (hunk 12 above)
Comment (imonroe) [reply modules/soe_paragraph_cta_list/templates/cta-list-paragraph.html.twig]: Per your feedback, I went with the approach which we used here: https://github.com/SU-SWS/jumpstart_ui/blob/8.x-1.x/templates/macros/field-links-to-list.twig

which prevents us from having to reference the field by name, but still lets us keep the logic in the template.

File: modules/soe_paragraph_image_cta/templates/image-cta-paragraph.html.twig (hunk 14 above)
Comment (imonroe) [reply modules/soe_paragraph_image_cta/templates/image-cta-paragraph.html.twig]: fixed.

File: modules/soe_paragraph_image_cta/templates/image-cta-paragraph.html.twig (hunk 15 above)
Comment (imonroe) [reply modules/soe_paragraph_image_cta/templates/image-cta-paragraph.html.twig]: done.

File: modules/soe_paragraph_stories/config/install/auto_entitylabel.settings.paragraph.soe_stories.yml (hunk 16 above)
Comment (imonroe) [reply modules/soe_paragraph_stories/config/install/auto_entitylabel.settings.paragraph.soe_stories.yml]: removed.

File: modules/soe_paragraph_stories/config/install/paragraphs.paragraphs_type.soe_stories.yml (hunk 17 above)
Comment (imonroe) [reply modules/soe_paragraph_stories/config/install/paragraphs.paragraphs_type.soe_stories.yml]: fixed.

File: modules/soe_paragraph_cta_list/templates/cta-list-paragraph.html.twig (hunk 13 above)
Comment (imonroe) [reply modules/soe_paragraph_cta_list/templates/cta-list-paragraph.html.twig]: consolidated.

File: modules/soe_paragraph_cta_list/config/install/core.entity_view_display.paragraph.cta_list.default.yml (hunk 10 above)
Comment (imonroe) [reply modules/soe_paragraph_cta_list/config/install/core.entity_view_display.paragraph.cta_list.default.yml]: Removed.

File: composer.json (hunk 9 above)
Comment (imonroe) [reply composer.json]: fixed.

File: modules/soe_paragraph_cta_list/config/install/paragraphs.paragraphs_type.cta_list.yml (hunk 8 above)
Comment (imonroe) [reply modules/soe_paragraph_cta_list/config/install/paragraphs.paragraphs_type.cta_list.yml]: I've renamed all the paragraphs to follow this convention.

File: modules/soe_paragraph_cta_list/config/install/core.entity_form_display.paragraph.cta_list.default.yml (hunk 7 above)
Comment (imonroe) [reply modules/soe_paragraph_cta_list/config/install/core.entity_form_display.paragraph.cta_list.default.yml]: UUIDs removed.

**Comment by sherakama** [comment]
@imonroe I think this is GTG. Fire away?
